### PR TITLE
chore: [4.4] update phpstan-baseline.php

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #7722~~

**Description**
Update phpstan-baseline.php for #7713
Fixes https://github.com/codeigniter4/CodeIgniter4/actions/runs/5617352335/job/15221317246

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [] Conforms to style guide

### CODE REVIEW

**Comment by paulbalandan** [comment]
Since the `4.4` branch is not yet released, can we instead add the type declarations/ PHPDocs to those new classes (HotReloader, SiteURI, etc.) and other additions?

File: phpstan-baseline.php [hunk 1]
```diff
@@ -1151,6 +1166,11 @@
 	'count' => 1,
 	'path' => __DIR__ . '/system/Exceptions/FrameworkException.php',
 ];
+$ignoreErrors[] = [
+	'message' => '#^Method CodeIgniter\\\\Exceptions\\\\FrameworkException\\:\\:forInvalidDirectory\\(\\) has no return type specified\\.$#',
+	'count' => 1,
+	'path' => __DIR__ . '/system/Exceptions/FrameworkException.php',
+];
```
Comment (kenjis) [inline phpstan-baseline.php:1173]: @paulbalandan The return type should be `static`, but we cannot use it yet:
https://github.com/codeigniter4/CodeIgniter4/blob/dec4b71ad42488043cb5955dafea22fc5aa3b6d5/system/Exceptions/FrameworkException.php#L36-L39

Comment (paulbalandan) [reply phpstan-baseline.php:1173]: Then a PHPDoc should suffice.

**Comment by kenjis** [comment]
Closed by #7723

**Pull request closed without merging**